src/coreclr/tools/cdac-build-tool/ObjectFileScraper.cs

```diff
@@ -387,6 +387,10 @@ private Content ReadContent(ScraperState state, HeaderDirectory header)
         state.ReadBytes(endMagic.AsSpan());
         if (!CheckEndMagic(endMagic))
         {
+            if (endMagic.All(b => b == 0))
+            {
+                throw new InvalidOperationException("expected endMagic, got all zeros. Did you add something to the data descriptor that can't be initialized at compile time?");
```

Comment:
What was it?

I wonder if we can do something like this:

```cxx
template <typename T>
T force_constexpr (T arg) constexpr
{
    return arg;
}
```

and then wrap every argument in all the `#define`s in `datadescriptor.cpp`  with `force_constexpr`:

```cxx
        /* .GlobalLiteralValues = */ {
#define CDAC_GLOBAL(name,tyname,value) { /*.Name = */ GET_GLOBAL_NAME(name), /* .TypeName = */ GET_GLOBALTYPE_NAME(name), /* .Value = */ force_constexpr (value) },
#include "datadescriptor.h"
        },

Reply:
The case that I hit was the `StressLog::maxMsgSize()` function not being constexpr. I ended up moving it to a constexpr field eventually to get the required behavior.

Reply:
C++20 will eventually give us `consteval` so we can tighten up the datadescriptor.cpp macros.  https://godbolt.org/z/8PG57rrzs

there are some awful hacks to make C++17 templates that would work but i'm not sure it's worth it

Reply:
We'd still need to update from C++11 first, which is already a struggle.